### DISCUSSION TITLE
Requires-Python upper bound ignored?

### DISCUSSION THREAD

**Discussion by pawamoy** [pdm-project/pdm#827] [General]
It seems my projects are not contaminated by upper bounds on Requires-Python set by dependencies (which is great). Or am I dreaming? If upper bounds really are ignored, I guess I missed the commit doing that? Could you share a link to such a commit?

All of this just out of curiosity!

**Comment by frostming** [comment]
What do you mean by "ignored"? If your project requires python `>=3.6,<4.0`, it will be written to the package metadata as-is. But the dependency solver will treat `<4.0` as no upper bound when resolving dependencies(There won't be a Python 4, GvR said). See this: https://github.com/pdm-project/pdm/search?q=self.MAX_MAJOR_VERSION

**Reply by pawamoy** [reply]
Yes I was talking about dependency resolution. Thanks, that's what I was looking for.
I'm actually very late to the party: [this comment](https://github.com/pdm-project/pdm/issues/454#issuecomment-836471552), later reinforced by [Henry Schreiner's article](https://iscinumpy.dev/post/bound-version-constraints/), led me to think that PDM considered the Requires-Python upper bound in *every case*, even for `<4.0`, when in fact it stopped doing that after https://github.com/pdm-project/pdm/issues/66.
